sap_hana/datadog_checks/sap_hana/sap_hana.py

```diff
@@ -0,0 +1,569 @@
+                'resource_type:{}'.format(volume['resource']),
+                'fs_path:{}'.format(volume['path']),
+            ]
+            tags.extend(self._tags)
+
+            host = self.get_hana_hostname(volume['host'])
+
+            # Read
+            reads = volume['reads']
+            self.gauge('volume.io.read.total', reads, tags=tags, hostname=host)
+            self.monotonic_count('volume.io.read.count', reads, tags=tags, hostname=host)
+
+            read_size = volume['read_size']
+            self.gauge('volume.io.read.size.total', read_size, tags=tags, hostname=host)
+            self.monotonic_count('volume.io.read.size.count', read_size, tags=tags, hostname=host)
+
+            read_percent = total_time_to_temporal_percent(volume['read_time'], scale=MICROSECOND)
+            self.rate('volume.io.read.utilized', read_percent, tags=tags, hostname=host)
+
+            # Write
+            writes = volume['writes']
+            self.gauge('volume.io.write.total', writes, tags=tags, hostname=host)
+            self.monotonic_count('volume.io.write.count', writes, tags=tags, hostname=host)
+
+            write_size = volume['write_size']
+            self.gauge('volume.io.write.size.total', write_size, tags=tags, hostname=host)
+            self.monotonic_count('volume.io.write.size.count', write_size, tags=tags, hostname=host)
+
+            write_percent = total_time_to_temporal_percent(volume['write_time'], scale=MICROSECOND)
+            self.rate('volume.io.write.utilized', write_percent, tags=tags, hostname=host)
+
+            # Overall
+
+            # Convert microseconds -> seconds
+            io_time = volume['io_time'] / MICROSECOND
+            io_percent = total_time_to_temporal_percent(io_time, scale=1)
+            self.rate('volume.io.utilized', io_percent, tags=tags, hostname=host)
+
+            if io_time:
+                throughput = (read_size + write_size) / io_time
+            else:
+                throughput = 0
+            self.gauge('volume.io.throughput', throughput, tags=tags, hostname=host)
+
+    def query_custom(self):
+        for custom_query in self._custom_queries:
+            query = custom_query.get('query')
+            if not query:  # no cov
+                self.log.error('Custom query field `query` is required')
+                continue
+
+            columns = custom_query.get('columns')
+            if not columns:  # no cov
+                self.log.error('Custom query field `columns` is required')
+                continue
+
+            self.log.debug('Running custom query for SAP HANA')
+            rows = self.iter_rows_raw(query)
+
+            # Trigger query execution
+            try:
+                first_row = next(rows)
+            except Exception as e:  # no cov
+                self.log.error('Error executing custom query: {}'.format(e))
+                continue
+
+            for row in chain((first_row,), rows):
+                if not row:  # no cov
```

Comment:
I'd be nice to cover those errors case, instead of marking them. Ideally we'd have at least a few tests with mocks, so that we can easily tests some simple bugs in the past.